# A System of Mutually Reliant Ratios Creates the Internal Consistency of the System

**Aether Cosmology Research Group.** Companion notebook to the PDF brief.
Works in the Chinese li and the metric metre.


## Preface: internal consistency

> "An ideal system of fundamental constants would be one in which these
> theoretical relationships were satisfied rigorously, while the adopted
> value of each individual constant agreed with its observed value."
> W. de Sitter & D. Brouwer (1938), p. 213. He adds: *our present system is
> not consistent.*

A distance is not measured directly. It is computed from an angle and a
known length, by trigonometry: a triangle needs **two of its three** parts
fixed. Feed in a radius R and an angle, get a distance. Change the unit of R
and every distance changes by the same factor. The unit is a free choice;
the geometry is not. We run the whole derivation in two units to show it.


# Part I: The Chinese System

## 1. The angle: the du

The Chinese divided one full turn of the sky into **365.25 du**, one du for
each day the Sun drifts around the sky in a year. One du is therefore a
little under one degree:

$$1\ \text{du} = \frac{360^\circ}{365.25} = 0.9856^\circ$$

Each du splits into 10 fen.


In [1]:
import math

# the du : the Chinese angle. one sky turn = 365.25 du (one per solar day-drift)
DU_PER_CIRCLE = 365.25
DEG_PER_DU    = 360.0 / DU_PER_CIRCLE
print(f"1 du = 360 / 365.25 = {DEG_PER_DU:.6f} deg = {DEG_PER_DU*60:.4f} arcmin")
print("1 du = 10 fen")


1 du = 360 / 365.25 = 0.985626 deg = 59.1376 arcmin
1 du = 10 fen


## 2. The length: the li

The li is the Chinese unit of ground distance ($1\ \text{li} = 300\ \text{bu}$).
In 724 AD the astronomer Yi Xing tied it to the sky: walking north along a
ground meridian, the pole star rises **1 du for every 351.267 li**
(Beer et al., 1961).

Closed into a full circle of 365.25 du, the survey gives the circumference of
the heavens in li, and the radius follows from $C = 2\pi R$:

$$\text{celestial circle} = 351.267 \times 365.25 \approx 128{,}300\ \text{li}
\qquad R = \frac{\text{circle}}{2\pi} \approx 20{,}420\ \text{li}$$

**Gai Tian cosmology.** This is the Chinese canopy-heaven model, *Gai Tian*
(Cullen, 1996; Needham, 1959): the ground is a flat plane and the li measures
travel across it; the du divides the sky, the Heavens turning around the
observer. The circle and radius above are dimensions of the celestial sphere,
not of a terrestrial globe. The arc is in the sky, a matter of optics; the
ground itself does not curve. The frame is observer-centric, each observer at
the centre of his own sky.


In [2]:
# the li : the Chinese length. Yi Xing (724 AD): pole star rises 1 du per 351.267 li
LI_PER_DU   = 351.267
meridian_li = LI_PER_DU * DU_PER_CIRCLE        # circumference of the heavens, in li
R_li        = meridian_li / (2 * math.pi)
print(f"celestial circle = 351.267 x 365.25 = {meridian_li:,.2f} li")
print(f"R_li             = circle / 2*pi    = {R_li:,.2f} li")


celestial circle = 351.267 x 365.25 = 128,300.27 li
R_li             = circle / 2*pi    = 20,419.62 li


## 3. The Moon and the Sun, in li

Parallax turns an angle into a distance. A body seen against the radius $R$
from section 2 subtends an angle $\theta$, and $d = R/\sin\theta$. Both
angles used here are parallax angles. $\theta_{\text{moon}}$ is the Moon's
horizontal parallax, the shift in its apparent place seen across a baseline
$R$. $\theta_{\text{sun}}$ is the solar parallax, fixed from timing a transit
of Venus or Mercury across the Sun's face.

$$d = \frac{R}{\sin\theta}$$

The Sun's distance is one AU; each planet's distance is its Kepler orbital
ratio times the AU.


In [3]:
# parallax law : d = R / sin(theta).  apply R_li to the measured parallax angles.
def parallax(R, angle_deg):
    return R / math.sin(math.radians(angle_deg))

theta_moon = 57.0       # lunar horizontal parallax, arcmin
theta_sun  = 8.794      # solar parallax, arcsec
moon_li = parallax(R_li, theta_moon / 60.0)
sun_li  = parallax(R_li, theta_sun / 3600.0)
print(f"Moon : theta_moon = 57'      d = R/sin = {moon_li:,.0f} li")
print(f"Sun  : theta_sun  = 8.794\"   d = R/sin = {sun_li:,.0f} li   (= 1 AU)")
print()
planets = [("Mercury",0.387098),("Venus",0.723332),("Earth",1.000000),
           ("Mars",1.523679),("Jupiter",5.202600),("Saturn",9.554910),
           ("Uranus",19.218400),("Neptune",30.110400)]
for name, a in planets:
    print(f"  {name:8} a = {a:8.4f} AU  ->  {a*sun_li:18,.0f} li")


Moon : theta_moon = 57'      d = R/sin = 1,231,591 li
Sun  : theta_sun  = 8.794"   d = R/sin = 478,945,812 li   (= 1 AU)

  Mercury  a =   0.3871 AU  ->         185,398,966 li
  Venus    a =   0.7233 AU  ->         346,436,832 li
  Earth    a =   1.0000 AU  ->         478,945,812 li
  Mars     a =   1.5237 AU  ->         729,759,676 li
  Jupiter  a =   5.2026 AU  ->       2,491,763,481 li
  Saturn   a =   9.5549 AU  ->       4,576,284,127 li
  Uranus   a =  19.2184 AU  ->       9,204,572,190 li
  Neptune  a =  30.1104 AU  ->      14,421,249,973 li


## 4. The speed of light and aberration, in li

The speed of light is a property of the vacuum, $c = 1/\sqrt{\mu_0\varepsilon_0}$.
Measured against li-based units it is about 959,800 li per second.

**Aberration route.** Earth's assumed motion is said to tilt starlight by the
aberration angle $\kappa$ (Bradley, 1728), with $\tan\kappa = v_\oplus/c$ and
$v_\oplus = 2\pi\,\text{AU}/T$, where $T$ is the year. Solved for the AU this
returns the Earth-Sun distance from $c$ and $\kappa$ alone, the route
Foucault ran in 1862 (Foucault, 1862).

**The aberration angle is fixed, not measured.** A telescope records one
quantity: the total stellar oscillation, the trepidation angle
$\kappa+N+p \approx 80''$. Nothing in the observation splits it into
aberration, nutation and precession. The route needs $\kappa$ to be exactly
the slice for which $c\tan\kappa$ equals the Kepler velocity
$2\pi\,\text{AU}/T$; that slice is not read off the oscillation, it is
assigned the value that closes the identity. In 1805 Laplace had Delambre
redetermine the light constant from the eclipses of Jupiter's moon Io, and
Delambre returned exactly Bradley's aberration figure (Laplace, 1805).
$\kappa$ and the solar parallax $\pi_\odot$ are both pinned to the AU, so the
aberration route and the parallax route of section 3 return the same
distance: that agreement is built in, not independent confirmation. In the
observer-centric Gai Tian frame the Earth does not move. The telescope records
one angle, the whole oscillation $\kappa+N+p$; the velocity read from it,
$c\tan(\kappa+N+p)$, is the speed of nothing.


In [4]:
# the speed of light : c = 1/sqrt(mu0*eps0), a property of the vacuum
EPS0 = 8.8541878128e-12        # F/m
MU0  = 1.25663706212e-6        # H/m
c_SI = 1.0 / math.sqrt(MU0 * EPS0)

# li in metres : the li circle (128,300 li) equals Earth's 40,075 km girth
factor = 40_075_000.0 / meridian_li        # metres per li
c_li   = c_SI / factor
print(f"c = 1/sqrt(mu0*eps0) = {c_li:,.0f} li/s")
print()

# aberration route, in li :  tan(kappa) = v/c ;  AU = v*T/2*pi ;  T = the year
kappa, N, prec = 20.49552, 9.2025, 50.2879
trepidation = kappa + N + prec
T_year      = 365.25636 * 86400.0
v_earth_li  = c_li * math.tan(math.radians(kappa / 3600.0))
AU_aberr_li = v_earth_li * T_year / (2 * math.pi)
lightmin_li = c_li * 60.0
print(f"T (the year)            = {T_year:,.0f} s")
print(f"v = c*tan(kappa)        = {v_earth_li:.3f} li/s")
print(f"1 AU = v*T/2*pi         = {AU_aberr_li:,.0f} li")
print(f"trepidation = kappa+N+p = {trepidation:.2f} arcsec")
print(f"parallax AU (section 3) = {sun_li:,.0f} li   gap = {(AU_aberr_li/sun_li-1)*100:+.3f}%")


c = 1/sqrt(mu0*eps0) = 959,787 li/s

T (the year)            = 31,558,150 s
v = c*tan(kappa)        = 95.369 li/s
1 AU = v*T/2*pi         = 479,005,190 li
trepidation = kappa+N+p = 79.99 arcsec
parallax AU (section 3) = 478,945,812 li   gap = +0.012%


# Part II: The Metric System

## 5. The angle and the metre

The metric angle is the **degree**: one full turn is 360 degrees.

The **metre** began in 1791 as $1/10{,}000{,}000$ of the Earth's meridian
quadrant, pole to equator, a length pegged to the pole star as the li is. The
Earth is an oblate spheroid; taking its equatorial circumference, 40,075 km,
the equatorial radius is

$$R = \frac{40{,}075\ \text{km}}{2\pi} \approx 6{,}378\ \text{km}$$

**On the radius.** The Earth is an oblate spheroid: equatorial radius
$a = 6{,}378.137$ km, polar radius $b = 6{,}356.752$ km. This brief uses the
equatorial radius. The commonly quoted 6,371 km is the IUGG mean radius
$R = (2a+b)/3$, an average over the spheroid.

**The two-sphere model.** The globe account drops the observer-centric sky
for a geocentric one. It places a fictitious observer at the centre of the
Earth; this linearises the separate observer hemispheres into one sphere and
sets a terrestrial sphere and the celestial sphere in 1:1 correspondence. The
radius here is that terrestrial sphere. Star positions and ground positions
are both referred to the single fictitious centre.


In [5]:
# the metre : Earth's equatorial circumference is 40,075 km (oblate spheroid)
circumference_m = 40_075_000.0
R_m             = circumference_m / (2 * math.pi)
print(f"degree: one turn = 360 deg")
print(f"equatorial circumference = {circumference_m:,.0f} m = 40,075 km")
print(f"R_m = circumference / 2*pi = {R_m:,.1f} m = {R_m/1000:,.2f} km")
print()
a_eq, b_pol = 6378.137, 6356.752               # equatorial, polar radius (km), WGS-84
R_iugg      = (2 * a_eq + b_pol) / 3
print(f"oblate Earth : a = {a_eq} km (equatorial),  b = {b_pol} km (polar)")
print(f"IUGG mean radius = (2a + b)/3 = {R_iugg:,.2f} km")


degree: one turn = 360 deg
equatorial circumference = 40,075,000 m = 40,075 km
R_m = circumference / 2*pi = 6,378,134.3 m = 6,378.13 km

oblate Earth : a = 6378.137 km (equatorial),  b = 6356.752 km (polar)
IUGG mean radius = (2a + b)/3 = 6,371.01 km


## 6. The Moon and the Sun, in metres

The same parallax law $d = R/\sin\theta$, the same observed angles, the
metric radius from section 5.


In [6]:
# same parallax law, same angles, metric R
moon_m = parallax(R_m, theta_moon / 60.0)
sun_m  = parallax(R_m, theta_sun / 3600.0)
print(f"Moon : d = {moon_m:,.0f} m = {moon_m/1000:,.0f} km")
print(f"Sun  : d = {sun_m:,.0f} m = {sun_m/1000:,.0f} km   (= 1 AU)")
print()
for name, a in planets:
    print(f"  {name:8} a = {a:8.4f} AU  ->  {a*sun_m/1000:16,.0f} km")


Moon : d = 384,691,499 m = 384,691 km
Sun  : d = 149,600,255,307 m = 149,600,255 km   (= 1 AU)

  Mercury  a =   0.3871 AU  ->        57,909,960 km
  Venus    a =   0.7233 AU  ->       108,210,652 km
  Earth    a =   1.0000 AU  ->       149,600,255 km
  Mars     a =   1.5237 AU  ->       227,942,767 km
  Jupiter  a =   5.2026 AU  ->       778,310,288 km
  Saturn   a =   9.5549 AU  ->     1,429,416,975 km
  Uranus   a =  19.2184 AU  ->     2,875,077,547 km
  Neptune  a =  30.1104 AU  ->     4,504,523,527 km


## 7. The speed of light and aberration, in metres

The same vacuum relation, in metric units. $\mu_0$ and $\varepsilon_0$ are
measured **per metre**, so $c = 1/\sqrt{\mu_0\varepsilon_0}$ comes out in
metres per second. The aberration route runs as in section 4, with $c$ in
m/s and $T$ the year.


In [7]:
# the same vacuum relation, in metric units : mu0, eps0 per metre -> c in m/s
print(f"c = 1/sqrt(mu0*eps0) = {c_SI:,.0f} m/s")
print()
v_earth_m  = c_SI * math.tan(math.radians(kappa / 3600.0))
AU_aberr_m = v_earth_m * T_year / (2 * math.pi)
print(f"v = c*tan(kappa) = {v_earth_m:,.1f} m/s")
print(f"1 AU = v*T/2*pi  = {AU_aberr_m:,.0f} m = {AU_aberr_m/1000:,.0f} km")


c = 1/sqrt(mu0*eps0) = 299,792,458 m/s

v = c*tan(kappa) = 29,788.9 m/s
1 AU = v*T/2*pi  = 149,618,802,181 m = 149,618,802 km


## 8. Light-time of flight, in metres

A distance can be restated as the time light takes to cross it. One
light-second is $c \times 1\,\text{s}$; one light-minute is sixty of those.
Dividing any distance by $c$ gives its light-time. This is the same metre
ladder, relabelled in seconds of light travel.


In [8]:
# light-time : a distance restated as the time light takes to cross it
lightsec_m = c_SI * 1.0
lightmin_m = c_SI * 60.0
print(f"1 light-second = {lightsec_m:,.0f} m = {lightsec_m/1000:,.0f} km")
print(f"1 light-minute = {lightmin_m:,.0f} m = {lightmin_m/1000:,.0f} km")
print(f"1 AU = {sun_m/c_SI/60:.2f} light-minutes")
print()
print("planet light-times (distance / c):")
for name, a in planets:
    print(f"  {name:8} {a*sun_m/c_SI/60:8.2f} min")


1 light-second = 299,792,458 m = 299,792 km
1 light-minute = 17,987,547,480 m = 17,987,547 km
1 AU = 8.32 light-minutes

planet light-times (distance / c):
  Mercury      3.22 min
  Venus        6.02 min
  Earth        8.32 min
  Mars        12.67 min
  Jupiter     43.27 min
  Saturn      79.47 min
  Uranus     159.84 min
  Neptune    250.42 min


## 9. The metre and c: an algebra switch

Until 1983 the metre was the standard and $c$ was measured against it. In
1983 $c$ was fixed at exactly 299,792,458 m/s and the metre was redefined as
the distance light travels in $1/c$ seconds. Light crosses one metre in
$1/c \approx 3.34$ ns; multiply that time back by $c$ and you recover one
metre. Only which quantity is held fixed changed. The length did not.


In [9]:
# 1983 : c fixed, metre rederived as the distance light goes in 1/c seconds
Nc       = 299_792_458
inv_c_ns = 1e9 / Nc
print(f"light crosses 1 m in 1/c = {inv_c_ns:.4f} ns")
print(f"c * (1/c) = {Nc * (1.0/Nc):.6f} m   ->  the metre does not move")


light crosses 1 m in 1/c = 3.3356 ns
c * (1/c) = 1.000000 m   ->  the metre does not move


# Part III: Conversion and Equivalence

## 10. Converting li to metres, and the summary

The li circle measures the heavens; the metric circle measures the
terrestrial sphere. The two-sphere model sets the two in 1:1 correspondence,
so the celestial circumference 128,300 li and Earth's circumference 40,075 km
describe one circle. Dividing gives the conversion factor:

$$1\ \text{li} = \frac{40{,}075{,}000\ \text{m}}{128{,}300\ \text{li}}
\approx 312.35\ \text{m}$$

Every distance is $R$ times an angle factor, and the angle factors are
identical in both systems, so every li value times 312.35 m/li reproduces the
metre value exactly.


In [10]:
# conversion factor, then every value in both units
print(f"1 li = 40,075,000 m / {meridian_li:,.2f} li = {factor:.4f} m = {factor/1000:.6f} km")
print()
rows = [
    ("Radius R",            R_li,        R_m),
    ("Moon distance",       moon_li,     moon_m),
    ("Sun distance (1 AU)", sun_li,      sun_m),
    ("AU via aberration",   AU_aberr_li, AU_aberr_m),
    ("1 light-minute",      lightmin_li, lightmin_m),
]
print(f"{'quantity':22}{'li':>18}{'metres':>20}{'li * factor':>20}")
for name, v_li, v_m in rows:
    print(f"{name:22}{v_li:18,.0f}{v_m:20,.0f}{v_li*factor:20,.0f}")
print()
print(f"speed of light : {c_li:,.0f} li/s  =  {c_SI:,.0f} m/s")


1 li = 40,075,000 m / 128,300.27 li = 312.3532 m = 0.312353 km

quantity                              li              metres         li * factor
Radius R                          20,420           6,378,134           6,378,134
Moon distance                  1,231,591         384,691,499         384,691,499
Sun distance (1 AU)          478,945,812     149,600,255,307     149,600,255,307
AU via aberration            479,005,190     149,618,802,181     149,618,802,181
1 light-minute                57,587,205      17,987,547,480      17,987,547,480

speed of light : 959,787 li/s  =  299,792,458 m/s


## 11. Scale invariance: R is the only free parameter

Every absolute quantity in the system carries a length dimension and is built
from R. The distances are R divided by the sine of an angle. The speed of
light, a length over a time, is locked to the AU through the aberration ring,
$c = 2\pi\,\text{AU}/(T\tan\kappa)$, so it moves with the AU.

Rescale R by a factor $k$ and the whole length scale moves with it: R, the
Moon, the AU, the planets, $c$, and the light-minute all scale by $k$. What
does **not** move: the angles ($du$, $\theta_{\text{moon}}$,
$\theta_{\text{sun}}$, $\kappa$), the periods, and the dimensionless ratios,
including the light-time AU/c.

The rescaling is shown below in Chinese li and in metric units. The $k = 1$
row is the accepted Earth, equatorial radius 6,378 km; it is one choice on a
free dial, and no angle in the sky fixes which choice is right.


In [11]:
# R is the system's one free scale. Rescale it by k; every length follows.
light_time_AU = sun_li / c_li          # AU / c : a time, invariant under rescaling

def rescaled(k):
    R    = k * R_li
    moon = parallax(R, theta_moon / 60.0)      # parallax distance, scales with R
    AU   = parallax(R, theta_sun / 3600.0)     # parallax distance, scales with R
    c    = AU / light_time_AU                  # c locked to the AU; light-time is the invariant
    return R, moon, AU, c

ks  = (0.25, 0.5, 1.0, 2.0, 4.0)
km  = factor / 1000.0                          # km per li

print("In Chinese li:")
print(f"{'k':>6}{'R (li)':>13}{'Moon (li)':>15}{'AU (li)':>18}{'c (li/s)':>14}")
for k in ks:
    R, moon, AU, c = rescaled(k)
    tag = "   <-- accepted Earth" if k == 1.0 else ""
    print(f"{k:>6}{R:>13,.0f}{moon:>15,.0f}{AU:>18,.0f}{c:>14,.0f}{tag}")

print()
print("In metric units (the same rows, li value x 0.31235 km/li):")
print(f"{'k':>6}{'R (km)':>12}{'Moon (km)':>14}{'AU (km)':>18}{'c (m/s)':>17}")
for k in ks:
    R, moon, AU, c = rescaled(k)
    tag = "   <-- accepted Earth" if k == 1.0 else ""
    print(f"{k:>6}{R*km:>12,.0f}{moon*km:>14,.0f}{AU*km:>18,.0f}{c*factor:>17,.0f}{tag}")

print()
print("k = 1 is the accepted Earth: R = 6,378 km, Moon = 384,691 km,")
print("AU = 1.496e8 km, c = 299,792,458 m/s. Every column scales exactly with k;")
print("the AU/c light-time (499 s) and all the angles stay fixed.")


In Chinese li:
     k       R (li)      Moon (li)           AU (li)      c (li/s)
  0.25        5,105        307,898       119,736,453       239,947
   0.5       10,210        615,796       239,472,906       479,893
   1.0       20,420      1,231,591       478,945,812       959,787   <-- accepted Earth
   2.0       40,839      2,463,183       957,891,624     1,919,573
   4.0       81,678      4,926,365     1,915,783,247     3,839,147

In metric units (the same rows, li value x 0.31235 km/li):
     k      R (km)     Moon (km)           AU (km)          c (m/s)
  0.25       1,595        96,173        37,400,064       74,948,115
   0.5       3,189       192,346        74,800,128      149,896,229
   1.0       6,378       384,691       149,600,255      299,792,458   <-- accepted Earth
   2.0      12,756       769,383       299,200,511      599,584,916
   4.0      25,513     1,538,766       598,401,021    1,199,169,832

k = 1 is the accepted Earth: R = 6,378 km, Moon = 384,691 km,
AU = 1.496

## Closing

The Moon, the Sun, the planets and the speed of light all come out in li and
in metres and convert into each other exactly. Neither unit is more correct.
A system of mutually reliant ratios is internally consistent by construction:
that is not the same as being independently verified.


## References

- Beer, A., Ho Ping-Yü, Lu Gwei-Djen, Needham, J., Pulleyblank, E. G., &
  Thompson, G. I. (1961). An 8th-century meridian line: I-Hsing's chain of
  gnomons and the pre-history of the metric system. *Vistas in Astronomy*, 4,
  3–28.
- Bradley, J. (1728). An account of a new discovered motion of the fixed
  stars. *Philosophical Transactions of the Royal Society*, 35, 637–661.
- Cullen, C. (1996). *Astronomy and Mathematics in Ancient China: The Zhou Bi
  Suan Jing*. Cambridge: Cambridge University Press.
- de Sitter, W., & Brouwer, D. (1938). On the system of astronomical
  constants. *Bulletin of the Astronomical Institutes of the Netherlands*,
  8(307), 213–231.
- Foucault, L. (1862). Détermination expérimentale de la vitesse de la
  lumière; parallaxe du Soleil. *Comptes Rendus de l'Académie des Sciences*,
  55, 501–503.
- Laplace, P.-S. (1805). *Traité de Mécanique Céleste, Tome IV: Théorie des
  satellites de Jupiter*. Paris: Courcier.
- Needham, J. (1959). *Science and Civilisation in China, Volume 3:
  Mathematics and the Sciences of the Heavens and the Earth*. Cambridge:
  Cambridge University Press.
